rest/src/main/kotlin/request/Request.kt

```diff
@@ -89,19 +90,13 @@ public class MultipartRequest<B : Any, R>(
         body?.let {
             append("payload_json", Json.encodeToString(it.strategy, it.body))
         }
-        try {
-
-            files.forEachIndexed { index, pair ->
-
-                val (name, inputStream) = pair
-                append(
-                    "file$index",
-                    inputStream.readBytes(),
-                    Headers.build { append(HttpHeaders.ContentDisposition, "filename=$name") }
-                )
-            }
-        } finally {
-            files.forEach { it.inputStream.close() }
+        files.forEachIndexed { index, pair ->
+            val (name, channel) = pair
+            append(
+                "file$index",
+                ChannelProvider { channel },
+                Headers.build { append(HttpHeaders.ContentDisposition, "filename=$name") }
+            )
```

Comment:
```suggestion
        files.forEachIndexed { index, (fileName, fileChannel) ->
            append(
                "file$index",
                ChannelProvider { fileChannel },
                headersOf(HttpHeaders.ContentDisposition, "filename=$fileName"),
            )
```
while we are here we can clean this up a bit